src/syrupy/extensions/amber/serializer.py

```diff
@@ -73,20 +73,21 @@ def read_file(cls, filepath: str) -> "SnapshotFossil":
         indent_len = len(cls._indent)
         snapshot_fossil = SnapshotFossil(location=filepath)
         try:
+            crn = "\r\n"
             with open(filepath, "r", encoding="utf-8", newline="") as f:
                 test_name = None
                 snapshot_data = ""
                 for line in f:
                     if line.startswith(cls._marker_name):
-                        test_name = line[name_marker_len:-1].strip(" \r\n")
+                        test_name = line[name_marker_len:-1].strip(f" {crn}")
                         snapshot_data = ""
                         continue
                     elif test_name is not None:
                         if line.startswith(cls._indent):
                             snapshot_data += line[indent_len:]
                         elif line.startswith(cls._marker_divider) and snapshot_data:
                             snapshot_fossil.add(
-                                Snapshot(name=test_name, data=snapshot_data[:-1])
+                                Snapshot(name=test_name, data=snapshot_data.rstrip(crn))
```

Comment:
This is not going to work / it won't be accurate.

Previously we only stripped the last character, which we _knew_ was a `\n` that we added and thus not part of the snapshot data. With `rstrip` it strips an arbitrary amount of carriage returns and newlines. It will even strip real data, if someone were to snapshot a newline at the end of their string.

We have full control over what we write in the snapshots, so I would stick with the "\n". No need to write carriage returns in snapshots for windows. Windows won't add carriage returns unless we tell it to.

Reply:
For amber this will be just before a new section is added to the snapshot, i.e. before the `---` which is terminated with a closing tag before the new line we add that's why this still works.

Reply:
@noahnu windows seems to behave differently when reading files that have newlines, at least I suspect that due the failing test logs. It would be better to have a consistent way of indicating in amber files that the snapshot data is ended and anything afterwards is a system new line and that can be applied and stipped regardless of the os newline characters.

Reply:
It shouldn't behave any differently since python handles the OS for us. I suspect something else is going on.

> Note Python doesn’t depend on the underlying operating system’s notion of text files; all the processing is done by Python itself, and is therefore platform-independent.
>
> &mdash; https://docs.python.org/3/library/functions.html#open

We specify how newlines are too be processed:

```
with open(filepath, "r", encoding="utf-8", newline="")
```

When `newline=""` we use universal line mode, so we read the exact same content on Windows / Mac / Linux.

Reply:
I would think the same thing but the logs seem to contradict and in the documentation (a little further down)
> If it is '', universal newlines mode is enabled, but line endings are returned to the caller untranslated
So it might be possible that the file on windows when read on windows has both `\r` and `\n` and because we use `""` not `None` the line endings are not translated into `\n` on input?